packages/react-components/lib/tasks/create-task.tsx

```diff
@@ -1244,33 +1295,42 @@ export function CreateTaskForm({
     }
     setScheduleUntilValue(event.target.value);
   };
-  // schedule is not supported with batch upload
-  const scheduleEnabled = taskRequests.length === 1;
 
   const [warnTimeChecked, setWarnTimeChecked] = React.useState(false);
   const handleWarnTimeCheckboxChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setWarnTimeChecked(event.target.checked);
   };
 
-  const updateTasks = () => {
-    setTaskRequests((prev) => {
-      prev.splice(selectedTaskIdx, 1, taskRequest);
-      return [...prev];
-    });
-  };
-
   const handleTaskDescriptionChange = (newDesc: TaskDescription) => {
-    taskRequest.category = 'compose';
-    taskRequest.description = newDesc;
+    setTaskRequest((prev) => {
+      return {
+        ...prev,
+        category: 'compose',
+        description: newDesc,
+      };
+    });
     setFavoriteTaskBuffer({ ...favoriteTaskBuffer, description: newDesc, category: 'compose' });
-    updateTasks();
   };
 
   const legacyHandleTaskDescriptionChange = (newCategory: string, newDesc: TaskDescription) => {
-    taskRequest.category = newCategory;
-    taskRequest.description = newDesc;
+    setTaskRequest((prev) => {
+      return {
+        ...prev,
+        category: newCategory,
+        description: newDesc,
+      };
+    });
     setFavoriteTaskBuffer({ ...favoriteTaskBuffer, description: newDesc, category: newCategory });
-    updateTasks();
+  };
+
+  const handleCustomTaskDescriptionChange = (newDesc: CustomTaskDescription) => {
```

Comment:
> Also is it intended that this does not update the favorite task buffer?

Yeah, I wasn't planning to support saving custom tasks as favorite tasks.

Since I'm re-visiting the task description changes, I'll see if saving it as a favorite task can be supported easily.

Reply:
To support favorite tasks for this custom compose, I'd probably like to properly define the model types, so I added a fixme for later

Reply:
Favorite task is probably the most useful with custom tasks, I think we should support it.

Reply:
Yes, but probably not at the moment. We will keep it in view until we finish the more critical items towards ongoing projects and release. For this PR, the cleanup to using single requests is required for the next few features